### PR TITLE
Northwest class 4-Humail Khan - JavaScript core 1-week 3 coursework

### PR DESCRIPTION
**Pull request by humailhasankhan**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

-----
[View rendered 1-exercises/C-array-every/README.md](https://github.com/humailhasankhan/JavaScript-Core-1-Coursework-Week3/blob/main/1-exercises/C-array-every/README.md)
[View rendered 1-exercises/D-array-filter/README.md](https://github.com/humailhasankhan/JavaScript-Core-1-Coursework-Week3/blob/main/1-exercises/D-array-filter/README.md)
[View rendered 1-exercises/E-array-map/README.md](https://github.com/humailhasankhan/JavaScript-Core-1-Coursework-Week3/blob/main/1-exercises/E-array-map/README.md)
[View rendered 1-exercises/F-array-forEach/README.md](https://github.com/humailhasankhan/JavaScript-Core-1-Coursework-Week3/blob/main/1-exercises/F-array-forEach/README.md)
[View rendered 1-exercises/H-array-methods-2/README.md](https://github.com/humailhasankhan/JavaScript-Core-1-Coursework-Week3/blob/main/1-exercises/H-array-methods-2/README.md)

### CODE REVIEW

File: 1-exercises/H-array-methods-2/README.md [hunk 1]
```diff
@@ -44,3 +44,5 @@ $ node
 ```
 
 There is a string method `.split()`. In an interactive console try using the string `.split()` method and the array `.join()`. How could they work together?
+
+Answer: if we choose an index from an array to split, it would log that string inside a separate array. If we join it back to the main array, it would join that string to the last index inside the main array where we join it.
```
Comment (dharmaguadeloupe) [inline 1-exercises/H-array-methods-2/README.md:48]: Good spot on answering this question. Somehow I missed that

File: 1-exercises/I-string-replace/exercise.js [hunk 2]
```diff
@@ -13,7 +13,13 @@
...
   "I like dogs. One day I went to the park and I saw 10 dogs. It was a great day.";
 
-let result = story.replace("", "");
+let result = story
+  .replace("dogs", "cats")
+  .replace("day", "night")
+  .replace("10", "100000")
+  .replace("dogs", "cats")
+  .replace("great", "brilliant")
+  .replace("day", "night");
```
Comment (dharmaguadeloupe) [inline 1-exercises/I-string-replace/exercise.js:22]: Great formatting for readability

File: 2-mandatory/1-create-functions.js [hunk 3]
```diff
@@ -44,7 +56,20 @@ Write a function that:
...
+      formattedNumber.push(`${number}%`);
+    } else if (number > 100) {
+      formattedNumber.push(`${Math.max(100)}%`);
+    } else if (number % 1 !== 0 && number > 1) {
+      formattedNumber.push(`${number.toFixed(1)}%`);
+    } else if (number % 1 !== 0 && number < 1) {
+      formattedNumber.push(`${number.toFixed(2)}%`);
+    }
+  });
+  return formattedNumber;
```
Comment (dharmaguadeloupe) [inline 2-mandatory/1-create-functions.js:72]: Different approach to how I tackled it. Very detailed

File: 1-exercises/H-array-methods-2/README.md (hunk 1 above)
Comment (humailhasankhan) [reply 1-exercises/H-array-methods-2/README.md:48]: Thank you Dharma.

File: 1-exercises/I-string-replace/exercise.js (hunk 2 above)
Comment (humailhasankhan) [reply 1-exercises/I-string-replace/exercise.js:22]: Thank you.

File: 2-mandatory/1-create-functions.js (hunk 3 above)
Comment (humailhasankhan) [reply 2-mandatory/1-create-functions.js:72]: It amazes me as well when I see other people's work. There are so many different ways to tackle one problem.

**Review by humailhasankhan — COMMENTED** [review]
The card validator in extra got pushed when I did "git add .". It is not complete and does not have a working solution. I just wanted to update my refactored code in the guardian's session for journey planner. Thank you.

**Pull request closed without merging**